### PR TITLE
Add replace text module

### LINKED ISSUE #155: Also check reports for issue URLs
## The Problem
[Clones #103]
Since #153 got merged already; #103 was not fully resolved by #153.
The functionality to remove links from the report itself is still not implemented, only comments are checked

## Possible Solution
Also check report body in the module added by #153

### PR DESCRIPTION
**Pull request by SPGoding**
## Purpose
Resolve #155.
## Approach
Renamed the old `ReplaceCommentModule` to `ReplaceTextModule`, and added check for issue description.
#### Checklist
- [x] Included tests
- [ ] Tested in MCTEST-xxx

### CODE REVIEW

File: src/main/kotlin/io/github/mojira/arisa/modules/Module.kt [hunk 1]
```diff
@@ -34,6 +34,13 @@ fun <T> assertNotNull(e: T?) = when (e) {
     else -> Unit.right()
 }
 
+fun assertOr(vararg list: Either<OperationNotNeededModuleResponse, ModuleResponse>) =
+    if (list.any { it.isRight() }) {
+        Unit.right()
+    } else {
+        OperationNotNeededModuleResponse.left()
+    }
```
Comment (NeunEinser) [inline src/main/kotlin/io/github/mojira/arisa/modules/Module.kt:42]: Nice!

File: src/main/kotlin/io/github/mojira/arisa/ModuleRegistry.kt [hunk 2]
```diff
@@ -377,6 +378,18 @@ class ModuleRegistry(jiraClient: JiraClient, private val config: Config) {
         }
 
         register(
+<<<<<<< HEAD
```
Comment (urielsalis) [inline src/main/kotlin/io/github/mojira/arisa/ModuleRegistry.kt:381]: seems you didnt merge correctly

Comment (SPGoding) [reply src/main/kotlin/io/github/mojira/arisa/ModuleRegistry.kt:381]: I'm working on that ;D

File: src/test/kotlin/io/github/mojira/arisa/modules/ReplaceTextModuleTest.kt [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.shouldBe
+
+class ReplaceTextModuleTest : StringSpec({
+    val module = ReplaceTextModule()
+    "should return OperationNotNeededModuleResponse when there is no description nor comment" {
+        val request = ReplaceTextModule.Request(
+            42L,
+            null,
+            listOf()
```
Comment (NeunEinser) [inline src/test/kotlin/io/github/mojira/arisa/modules/ReplaceTextModuleTest.kt:15]: [Minor] can be `emptyList()`

File: src/test/kotlin/io/github/mojira/arisa/modules/ReplaceTextModuleTest.kt [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+import arrow.core.right
+import io.kotest.assertions.arrow.either.shouldBeLeft
+import io.kotest.assertions.arrow.either.shouldBeRight
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.shouldBe
+
+class ReplaceTextModuleTest : StringSpec({
+    val module = ReplaceTextModule()
+    "should return OperationNotNeededModuleResponse when there is no description nor comment" {
+        val request = ReplaceTextModule.Request(
```
Comment (NeunEinser) [inline src/test/kotlin/io/github/mojira/arisa/modules/ReplaceTextModuleTest.kt:12]: [Minor] I generally like to add imports for these so you can simply use `Request`, or `Comment` (Alt + Enter, add Import)

**Pull request merged**